src/app/core/auth/models/auth-method.model.ts

```diff
@@ -0,0 +1,44 @@
+export class AuthMethodModel {
+  authMethodName: string;
+  location?: string;
+  authMethodConstant: AuthMethodConstants
+
+  constructor(authMethodName: string, location?: string) {
+    this.authMethodName = authMethodName;
+    this.location = location;
+    switch (authMethodName) {
+      case 'ip': {
+        this.authMethodConstant = AuthMethodConstants.IP;
+        break;
+      }
+      case 'ldap': {
+        this.authMethodConstant = AuthMethodConstants.LDAP;
+        break;
+      }
+      case 'shibboleth': {
+        this.authMethodConstant = AuthMethodConstants.SHIBBOLETH;
+        break;
+      }
+      case 'x509': {
+        this.authMethodConstant = AuthMethodConstants.X509;
+        break;
+      }
+      case 'password': {
+        this.authMethodConstant = AuthMethodConstants.PASSWORD;
+        break;
+      }
+
+      default: {
+        break;
+      }
+    }
+  }
+}
+
+export enum AuthMethodConstants {
```

Comment:
please use string enum so you can easily map with auth method name 
```
export enum AuthMethodType {
  Password = 'password',
  Shibboleth = 'shibboleth',
  Ldap = 'ldap',
  Ip = 'ip'
}
```

Reply:
Changed it as suggested